torchhd/datasets/abalone.py

```diff
@@ -0,0 +1,165 @@
+import os
+import os.path
+from typing import Callable, Optional, Tuple, List
+import torch
+from torch.utils import data
+import pandas as pd
+import tarfile
+import numpy as np
+
+from .utils import download_file
+
+
+class Abalone(data.Dataset):
+    """`Abalone <https://archive.ics.uci.edu/ml/datasets/abalone>`_ dataset.
+
+    Args:
+        root (string): Root directory containing the files of the dataset.
+        train (bool, optional): If True, returns training (sub)set stored in ``train.data`` as further determined by fold_id and fold_train variables.
+            Otherwise tries to return test set if ``test.data`` exists, issues error in case it is not available.
+        fold_id (int, optional): Specifies which fold number to use. Relevant only if train is set to True. The default value of 0 returns the whole data in ``train.data``.
```

Comment:
I like that you can still get all the data if desired, maybe we should change the API to return everything when the fold is -1 and otherwise it returns the fold between 0 and 3 such that in your python code you can do:
```python
for fold in range(4):
    pass
```

Right now this could lead to unintended hard to catch bugs because 0 returns everything. Which makes me thing adding a class propriety `num_folds` would be helpful as well.